src/webui/www/private/speedlimit.html

```diff
@@ -0,0 +1,154 @@
+<!DOCTYPE html>
+<html lang="${LANG}" class="dark">
+
+<head>
+    <meta charset="UTF-8">
+    <title>QBT_TR(Speed limit)QBT_TR[CONTEXT=SpeedLimitDialog]</title>
+    <link rel="stylesheet" href="css/style.css?v=${CACHEID}" type="text/css">
+    <script src="scripts/localpreferences.js?v=${CACHEID}"></script>
+    <script src="scripts/color-scheme.js?v=${CACHEID}"></script>
+</head>
+
+<body>
+    <div style="padding-top: 10px; width: 100%; text-align: center; margin: 0 auto; overflow: hidden">
+        <div class="slider">
+            <div class="update">
+                <label id="limitUpdateLabel" for="limitUpdateValue">QBT_TR(Limit:)QBT_TR[CONTEXT=SpeedLimitDialog]</label>
+                <input type="text" id="limitUpdateValue" size="6" placeholder="∞" style="text-align: center;">
+                <span id="limitUnit">QBT_TR(KiB/s)QBT_TR[CONTEXT=SpeedLimitDialog]</span>
+            </div>
+            <input type="range" id="limitSliderInput" value="0" style="width: 100%;" aria-label="QBT_TR(Download limit)QBT_TR[CONTEXT=PropertiesWidget]">
+        </div>
+        <input type="button" id="applyButton" value="QBT_TR(Apply)QBT_TR[CONTEXT=HttpServer]" onclick="setLimit()">
+    </div>
+
+    <script>
+        "use strict";
+
+        window.addEventListener("keydown", (event) => {
+            switch (event.key) {
+                case "Enter":
+                    event.preventDefault();
+                    document.getElementById("applyButton").click();
+                    break;
+                case "Escape":
+                    event.preventDefault();
+                    window.parent.qBittorrent.Client.closeFrameWindow(window);
+                    break;
+            }
+        });
+
+        const params = new URLSearchParams(window.location.search);
+        const type = params.get("type");
+        const hashes = params.get("hashes").split("|");
+
+        document.getElementById("limitUpdateLabel").textContent =
+            type === "upload"
+            ? `QBT_TR(Upload limit:)QBT_TR[CONTEXT=PropertiesWidget]`
+            : `QBT_TR(Download limit:)QBT_TR[CONTEXT=PropertiesWidget]`;
+
+        const setupSlider = (limit, maximum) => {
+            const input = document.getElementById("limitSliderInput");
+            input.setAttribute("max", maximum);
+            input.setAttribute("min", 0);
+            input.value = limit;
+            input.addEventListener("input", (event) => {
+                const pos = Number(event.target.value);
+                if (pos > 0) {
+                    document.getElementById("limitUpdateValue").value = pos;
+                    document.getElementById("limitUnit").style.visibility = "visible";
+                }
+                else {
+                    document.getElementById("limitUpdateValue").value = "∞";
+                    document.getElementById("limitUnit").style.visibility = "hidden";
```

Comment:
It would be nice to have variables to them in order to avoid repeating `document.getElementById` everywhere. For example:
```js
"use strict";

const limitUpdateValue = document.getElementById("limitUpdateValue");
const limitUnit = document.getElementById("limitUnit");

// reuse them
limitUpdateValue.value = "∞";
```